virtcontainers/container.go

```diff
@@ -300,6 +300,38 @@ func (c *Container) createContainersDirs() error {
 	return nil
 }
 
+func (c *Container) attachBlockForMount(m Mount) (*drivers.BlockDevice, error) {
+	var stat unix.Stat_t
+	if err := unix.Stat(m.Source, &stat); err != nil {
+		return nil, err
+	}
+
+	// Check if mount is a block device file. If it is, the block device will be attached to the host
+	// instead of passing this as a shared mount.
+	if c.checkBlockDeviceSupport() && stat.Mode&unix.S_IFBLK == unix.S_IFBLK {
+		b, err := c.sandbox.devManager.NewDevice(config.DeviceInfo{
+			HostPath:      m.Source,
+			ContainerPath: m.Destination,
+			DevType:       "b",
+			Major:         int64(stat.Rdev / 256),
+			Minor:         int64(stat.Rdev % 256),
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// Attach this block device, all other devices passed in the config have been attached at this point
+		if err := b.Attach(c.sandbox); err != nil {
+			return nil, err
+		}
+
+		// TODO: remove dependency of package drivers
```

Comment:
If this TODO is going to be retained, please could you change it to reference a github issue so we don't forget to do this.

Same comment for the other TODO's.

Reply:
Sure

Reply:
Addressed in https://github.com/kata-containers/runtime/issues/306 and self-assigned.
Will handle this soon